utils/post_processing/restructure_borealis.py

```diff
@@ -0,0 +1,507 @@
+				'data_descriptors', 'data_normalization_factor', 'experiment_comment', 'experiment_id',
+				'experiment_name', 'first_range', 'first_range_rtt', 'freq', 'intf_antenna_count', 'lags',
+				'main_antenna_count', 'num_ranges', 'num_samps', 'num_slices', 'pulse_phase_offset', 'pulses',
+				'range_sep', 'rx_sample_rate', 'samples_data_type', 'scan_start_marker', 'slice_comment',
+				'station', 'tau_spacing', 'tx_pulse_len']
+
+shared_rawacf = ['beam_azms', 'beam_nums', 'blanked_samples', 'borealis_git_hash', 'correlation_descriptors',
+				 'data_normalization_factor', 'experiment_comment', 'experiment_id', 'experiment_name',
+				 'first_range', 'first_range_rtt', 'freq', 'intf_antenna_count', 'lags', 'main_antenna_count',
+				 'num_slices', 'pulses', 'range_sep', 'rx_sample_rate', 'samples_data_type', 'scan_start_marker',
+				 'slice_comment', 'station', 'tau_spacing', 'tx_pulse_len', 'correlation_dimensions']
+
+def find_max_sequences(data):
+	"""
+	Finds the maximum number of sequences between records in a Borealis data file
+	Args
+		data:		Site formatted data from a Borealis file, 
+					organized as one record for each slice
+	Returns:
+		max_seqs:	The largest number of sequences found 
+					in one record from the file
+	"""
+	first_key = list(data.keys())[0]
+	max_seqs = data[first_key]["num_sequences"]
+	for k in data:
+		if max_seqs < data[k]["num_sequences"]:
+			max_seqs = data[k]["num_sequences"]
+	return max_seqs
+
+
+def antennas_iq_site_to_array(data_record):
+	"""
+	Restructuring method for pre bfiq data
+	args:
+		data_record		a timestamped record loaded from an
+		 				hdf5 Borealis pre-bfiq data file
+		 Returns:
+		 	data_dict:	A dictionary containing the data from data_record
+		 				reformatted to be stored entirely as arrays, or as
+		 				one entry if the field does not change between records
+	"""
+	data_dict = dict()
+	num_records = len(data_record)
+
+	# write shared fields to dictionary
+	k = list(data_record.keys())[0]
+	for field in shared_antiq:
+		data_dict[field] = data_record[k][field]
+
+	# find maximum number of sequences
+	max_seqs = find_max_sequences(data_record)
+	dims = data_record[k]["data_dimensions"]
+	num_antennas = dims[0]
+	num_samps = dims[2]
+
+	data_buffer_offset = num_antennas * num_samps * max_seqs
+	data_buffer = np.zeros(num_records * data_buffer_offset, dtype=np.complex64)
+	data_shape = (num_records, num_antennas, max_seqs, num_samps)
+
+	sqn_buffer_offset = max_seqs
+	sqn_ts_buffer = np.zeros(num_records * max_seqs)
+	sqn_shape = (num_records, max_seqs)
+
+	noise_buffer_offset = max_seqs
+	noise_buffer = np.zeros(num_records * max_seqs)
+	noise_shape = (num_records, max_seqs)
+
+	sqn_num_array = np.empty(num_records)
+	int_time_array = np.empty(num_records)
+
+	data_dims_array = np.empty((num_records, len(data_record[k]["data_descriptors"])))
+
+	rec_idx = 0
+	for k in data_record:
```

Comment:
depends upon being an ordered dict, right? So we should maybe say this needs to run under python 3.6 or higher if that's the case. You can do a check of the python version and print an error if it's not high enough: https://docs.python.org/2/library/sys.html#sys.version_info

Reply:
My understanding was that this is good for iterating over regular dictionaries in Python as well.

Reply:
Yes, it should work to iterate through either type of dictionary, but I'm referring to the fact you're using k (indirectly) as an index into some arrays (int_time_array, seq_num_array, etc...) by incrementing rec_idx on every run through the loop. Do these arrays need to be ordered? If not, then no worries, but What if the dictionary isn't ordered and the arrays do need to be ordered?

Reply:
Ah I think I see what you mean. My intention was to have the data in the arrays stored in the same order as they were in the original record based format. Since the keys to the dictionary are the timestamps of the first sequence in the record I assumed that they would already be in the right order in the dictionary.

Reply:
Perfect! I didn't know for certain.